addons/crm/views/crm_lead_views.xml

```diff
@@ -24,6 +24,7 @@
                     </header>
                     <sheet>
                         <field name="active" invisible="1"/>
+                        <widget name="web_ribbon" title="Archived" bg_color="bg-danger" attrs="{'invisible': [('active', '=', True)]}"/>
```

Comment:
There is a "Lost" ribbon that may conflict with this one (active=False, probability 0 -> two ribbons normally).